guild.go

```diff
@@ -137,66 +110,40 @@ func (c *Client) ModifyGuild(id objects.Snowflake, params *ModifyGuildParams) (*
 		reason = params.Reason
 	}
 
-	res, err := c.request(&request{
-		method:      http.MethodPatch,
-		path:        fmt.Sprintf(GuildBaseFmt, id),
-		contentType: JsonContentType,
-		body:        data,
-		reason:      reason,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	if err = res.ExpectsStatus(http.StatusOK); err != nil {
-		return nil, err
-	}
-
 	guild := &objects.Guild{}
-	if err = res.JSON(guild); err != nil {
-		return nil, err
-	}
+	err = NewRequest().
+		Method(http.MethodPatch).
+		Path(fmt.Sprintf(GuildBaseFmt, id)).
+		ContentType(JsonContentType).
+		Body(data).
+		Reason(reason).
+		Expect(http.StatusOK).
+		Bind(guild).
+		Send(c)
 
-	return guild, nil
+	return guild, err
 }
 
 func (c *Client) DeleteGuild(id objects.Snowflake) error {
-	res, err := c.request(&request{
-		method:      http.MethodDelete,
-		path:        fmt.Sprintf(GuildBaseFmt, id),
-		contentType: JsonContentType,
-	})
-	if err != nil {
-		return err
-	}
-
-	if err = res.ExpectsStatus(http.StatusNoContent); err != nil {
-		return err
-	}
-
-	return nil
+	return NewRequest().
+		Method(http.MethodDelete).
+		Path(fmt.Sprintf(GuildBaseFmt, id)).
+		ContentType(JsonContentType).
+		Expect(http.StatusNoContent).
+		Send(c)
 }
 
 func (c *Client) GetGuildChannels(id objects.Snowflake) ([]*objects.Channel, error) {
-	res, err := c.request(&request{
-		method:      http.MethodGet,
-		path:        fmt.Sprintf(GuildChannelsFmt, id),
-		contentType: JsonContentType,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	if err = res.ExpectsStatus(http.StatusOK); err != nil {
-		return nil, err
-	}
-
-	var channels []*objects.Channel
-	if err = res.JSON(&channels); err != nil {
-		return nil, err
-	}
+	channels := []*objects.Channel{}
+	err := NewRequest().
+		Method(http.MethodGet).
+		Path(fmt.Sprintf(GuildChannelsFmt, id)).
+		ContentType(JsonContentType).
+		Expect(http.StatusOK).
+		Bind(channels).
```

Comment:
We need to pass a pointer to the slice here, otherwise it will automatically return an error.